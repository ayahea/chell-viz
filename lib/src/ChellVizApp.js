"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var React = require("react");
var semantic_ui_react_1 = require("semantic-ui-react");
var chell_1 = require("chell");
var VizPanelContainer_1 = require("./container/VizPanelContainer");
var ChellVizApp = /** @class */ (function (_super) {
    __extends(ChellVizApp, _super);
    function ChellVizApp() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ChellVizApp.prototype.render = function () {
        return (React.createElement("div", { id: "ChellVizApp" },
            React.createElement(semantic_ui_react_1.Grid, { centered: true, divided: 'vertically' },
                React.createElement(semantic_ui_react_1.GridRow, null,
                    React.createElement(VizPanelContainer_1.VizPanelContainer, { dataDirs: ['1', '2', '3'].map(function (dir) { return "assets/contact_map/example" + dir; }), supportedVisualizations: [chell_1.VIZ_TYPE.CONTACT_MAP, chell_1.VIZ_TYPE.NGL], initialVisualizations: [chell_1.VIZ_TYPE.CONTACT_MAP, chell_1.VIZ_TYPE.NGL], numPanels: 2 })),
                React.createElement(semantic_ui_react_1.GridRow, null,
                    React.createElement(VizPanelContainer_1.VizPanelContainer, { dataDirs: ['centroids', 'centroids_subset', 'ngl', 'spring2/full'].map(function (dir) { return "assets/" + dir; }), initialVisualizations: [chell_1.VIZ_TYPE['T-SNE'], chell_1.VIZ_TYPE.SPRING], supportedVisualizations: [chell_1.VIZ_TYPE['T-SNE'], chell_1.VIZ_TYPE.SPRING], numPanels: 2 })))));
    };
    return ChellVizApp;
}(React.Component));
exports.ChellVizApp = ChellVizApp;
