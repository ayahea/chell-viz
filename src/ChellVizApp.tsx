import * as React from 'react';
import { Grid, GridRow } from 'semantic-ui-react';

import { VIZ_TYPE } from './component/VizSelectorPanel';
import { VizPanelContainer } from './container/VizPanelContainer';

export class ChellVizApp extends React.Component<any, any> {
  public render() {
    return (
      <div id="ChellVizApp">
        <Grid centered={true} divided={'vertically'}>
          <GridRow>
            <VizPanelContainer initialVisualizations={[VIZ_TYPE.CONTACT_MAP, VIZ_TYPE.NGL]} numPanels={2} />
          </GridRow>
        </Grid>
        <GridRow>
          <VizPanelContainer initialVisualizations={[VIZ_TYPE['T-SNE'], VIZ_TYPE.SPRING]} numPanels={2} />
        </GridRow>
      </div>
    );
  }
}
