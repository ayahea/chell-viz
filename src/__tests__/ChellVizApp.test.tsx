import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import * as React from 'react';

import { ChellVizApp } from '../ChellVizApp';

describe('ChellVizApp', () => {
  it('Should match existing snapshot when given no props.', () => {
    expect(toJson(shallow(<ChellVizApp />))).toMatchSnapshot();
  });
});
