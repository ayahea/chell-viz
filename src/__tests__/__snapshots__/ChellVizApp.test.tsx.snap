// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChellVizApp Should match existing snapshot when given no props. 1`] = `
<div
  id="ChellVizApp"
>
  <Grid
    centered={true}
    divided="vertically"
  >
    <GridRow>
      <VizPanelContainerClass
        dataDirs={
          Array [
            "assets/contact_map/example1",
            "assets/contact_map/example2",
            "assets/contact_map/example3",
          ]
        }
        initialVisualizations={
          Array [
            "Contact Map",
            "NGL",
            "Contact Map",
          ]
        }
        numPanels={3}
        supportedVisualizations={
          Array [
            "Contact Map",
            "NGL",
          ]
        }
      />
    </GridRow>
    <GridRow>
      <VizPanelContainerClass
        dataDirs={
          Array [
            "assets/centroids",
            "assets/centroids_subset",
            "assets/ngl",
            "assets/spring2/full",
          ]
        }
        initialVisualizations={
          Array [
            "T-SNE",
            "Spring",
          ]
        }
        numPanels={2}
        supportedVisualizations={
          Array [
            "T-SNE",
            "Spring",
          ]
        }
      />
    </GridRow>
  </Grid>
</div>
`;
